cmd/cronosd/cmd/gentx.go

```diff
@@ -0,0 +1,535 @@
+			if err != nil {
+				return errors.Wrap(err, "failed to build create-validator message")
+			}
+
+			ethSig, err := hexutil.Decode(args[4])
+			if err != nil {
+				return err
+			}
+
+			delegateGravityMsg := &gravitytypes.MsgDelegateKeys{
+				ValidatorAddress:    sdk.ValAddress(key.GetAddress()).String(),
+				OrchestratorAddress: orchAddress.String(),
+				EthereumAddress:     ethAddress,
+				EthSignature:        ethSig,
+			}
+
+			msgs := []sdk.Msg{msg, delegateGravityMsg}
+
+			if key.GetType() == keyring.TypeOffline || key.GetType() == keyring.TypeMulti {
+				cmd.PrintErrln("Offline key passed in. Use `tx sign` command to sign.")
+				return authclient.PrintUnsignedStdTx(txBldr, clientCtx, msgs)
+			}
+
+			// write the unsigned transaction to the buffer
+			w := bytes.NewBuffer([]byte{})
+			clientCtx = clientCtx.WithOutput(w)
+
+			if err = authclient.PrintUnsignedStdTx(txBldr, clientCtx, msgs); err != nil {
+				return errors.Wrap(err, "failed to print unsigned std tx")
+			}
+
+			// read the transaction
+			stdTx, err := readUnsignedGenTxFile(clientCtx, w)
+			if err != nil {
+				return errors.Wrap(err, "failed to read unsigned gen tx file")
+			}
+
+			// sign the transaction and write it to the output file
+			txBuilder, err := clientCtx.TxConfig.WrapTxBuilder(stdTx)
+			if err != nil {
+				return fmt.Errorf("error creating tx builder: %w", err)
+			}
+
+			err = authclient.SignTx(txFactory, clientCtx, name, txBuilder, true, true)
+			if err != nil {
+				return errors.Wrap(err, "failed to sign std tx")
+			}
+
+			outputDocument, _ := cmd.Flags().GetString(flags.FlagOutputDocument)
+			if outputDocument == "" {
+				outputDocument, err = makeOutputFilepath(config.RootDir, nodeID)
+				if err != nil {
+					return errors.Wrap(err, "failed to create output file path")
+				}
+			}
+
+			if err := writeSignedGenTx(clientCtx, outputDocument, stdTx); err != nil {
+				return errors.Wrap(err, "failed to write signed gen tx")
+			}
+
+			cmd.PrintErrf("Genesis transaction written to %q\n", outputDocument)
+			return nil
+		},
+	}
+
+	cmd.Flags().String(flags.FlagHome, defaultNodeHome, "The application home directory")
+	cmd.Flags().String(flags.FlagOutputDocument, "", "Write the genesis transaction JSON document to the given file instead of the default location")
+	cmd.Flags().String(flags.FlagChainID, "", "The network chain ID")
+	cmd.Flags().AddFlagSet(fsCreateValidator)
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func makeOutputFilepath(rootDir, nodeID string) (string, error) {
+	writePath := filepath.Join(rootDir, "config", "gentx")
+	if err := tmos.EnsureDir(writePath, 0700); err != nil {
+		return "", err
+	}
+
+	return filepath.Join(writePath, fmt.Sprintf("gentx-%v.json", nodeID)), nil
+}
+
+func readUnsignedGenTxFile(clientCtx client.Context, r io.Reader) (sdk.Tx, error) {
+	bz, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	aTx, err := clientCtx.TxConfig.TxJSONDecoder()(bz)
+	if err != nil {
```

Comment:
might just `return aTx, err` directly instead of check `err` here.

Reply:
fixed